### PR TITLE
Compat eoy

### PR DESCRIPTION
**Pull request by piatra**

### CODE REVIEW

File: test/unit/asrouter/compatibility-reference/snippets-fx57.js [hunk 1]
```diff
@@ -95,4 +95,28 @@ export const expectedValues = {
     error_text: "error",
     success_text: "all good",
   },
+
+  // EOY Snippet (https://github.com/mozmeao/snippets/blob/master/activity-stream/mofo-eoy-2017.html)
+  eoy_snippet: {
+    donation_form_url: "https://donate.mozilla.org/",
```
Comment (k88hudson) [inline test/unit/asrouter/compatibility-reference/snippets-fx57.js]: should have `block_button_text`

Comment (k88hudson) [reply test/unit/asrouter/compatibility-reference/snippets-fx57.js]: should have `selected_button`

**Comment by k88hudson** [comment]
Looks good after rebase, small nits to match schema

File: content-src/asrouter/components/SnippetBase/SnippetBase.jsx [hunk 2]
```diff
@@ -30,7 +30,7 @@ export class SnippetBase extends React.PureComponent {
     }
 
     return (
-      <button className="blockButton" title={this.props.content.block_button_text} onClick={this.onBlockClicked} />
+      <button className="blockButton" title={this.props.content.block_button_text || "Remove this"} onClick={this.onBlockClicked} />
```
Comment (k88hudson) [inline content-src/asrouter/components/SnippetBase/SnippetBase.jsx]: This should be localized eventually

Comment (piatra) [reply content-src/asrouter/components/SnippetBase/SnippetBase.jsx]: https://bugzilla.mozilla.org/show_bug.cgi?id=1500489

**Pull request merged**